src/main/java/com/iexec/core/replicate/ReplicatesHelper.java

```diff
@@ -0,0 +1,131 @@
+
+import static com.iexec.common.replicate.ReplicateStatus.*;
+
+public class ReplicatesHelper {
+    private ReplicatesHelper() {
+
+    }
+
+    /**
+     * Computes the number of replicates in the {@link ReplicateStatus#CONTRIBUTED} status
+     * that have the right contribution hash.
+     * <p>
+     * Note this method won't retrieve the replicates list, so it could be static.
+     * For test purposes - i.e. mocking -, it is not.
+     *
+     * @param replicates       List of replicates on which the calculation should be base.
+     * @param contributionHash Valid hash of the final result.
+     * @return Number of winners who had contributed with the right hash.
+     */
+    public static int getNbValidContributedWinners(List<Replicate> replicates, String contributionHash) {
+        int nbValidWinners = 0;
+        for (Replicate replicate : replicates) {
+            Optional<ReplicateStatus> oStatus = replicate.getLastRelevantStatus();
+            if (oStatus.isPresent() && oStatus.get().equals(CONTRIBUTED)
+                    && contributionHash.equals(replicate.getContributionHash())) {
+                nbValidWinners++;
+            }
+        }
+        return nbValidWinners;
+    }
+
+    public static int getNbReplicatesWithCurrentStatus(List<Replicate> replicates, ReplicateStatus... listStatus) {
+        int nbReplicates = 0;
+        for (Replicate replicate : replicates) {
+            for (ReplicateStatus status : listStatus) {
+                if (replicate.getCurrentStatus().equals(status)) {
+                    nbReplicates++;
+                }
+            }
+        }
+        return nbReplicates;
+    }
+
+    public static int getNbReplicatesWithLastRelevantStatus(List<Replicate> replicates, ReplicateStatus... listStatus) {
+        int nbReplicates = 0;
+        for (Replicate replicate : replicates) {
+            for (ReplicateStatus status : listStatus) {
+                if (Objects.equals(replicate.getLastRelevantStatus().orElse(null), status)) {
+                    nbReplicates++;
+                }
+            }
+        }
+        return nbReplicates;
+    }
+
+    public static int getNbReplicatesContainingStatus(List<Replicate> replicates, ReplicateStatus... listStatus) {
+        Set<String> addressReplicates = new HashSet<>();
+        for (Replicate replicate : replicates) {
+            List<ReplicateStatus> listReplicateStatus = replicate.getStatusUpdateList().stream()
+                    .map(ReplicateStatusUpdate::getStatus)
+                    .collect(Collectors.toList());
+            for (ReplicateStatus status : listStatus) {
+                if (listReplicateStatus.contains(status)) {
+                    addressReplicates.add(replicate.getWalletAddress());
+                }
+            }
+        }
+        return addressReplicates.size();
+    }
+
+    static boolean isStatusBeforeWorkerLostEqualsTo(Replicate replicate, ReplicateStatus status) {
```

Comment:
Wouldn't be better to put this method in `Replicate`?

Reply:
Sure!